build.cake

```diff
@@ -91,17 +91,17 @@ Task("Build")
     .IsDependentOn("Clean")
     .Does(() =>
     {
-        if (IsRunningOnLinux())
-            BuildEachProjectSeparatelyOnLinux();
+        if (IsRunningOnWindows())
+            BuildSolution();
         else
-            BuildUsingMSBuildOnWindowsOrMacOS();
+            BuildEachProjectSeparately();
     });
 
-public void BuildUsingMSBuildOnWindowsOrMacOS()
+public void BuildSolution()
 {
     MSBuild(SOLUTION_FILE, CreateMSBuildSettings("Build").WithRestore());
 
-    Information("Publishing .NET Core & Standard projects so that dependencies are present...");
+    DisplayBanner("Publish .NET Core & Standard projects");
 
     foreach (var framework in new[] { "netstandard2.0", "netcoreapp3.1" })
         MSBuild(ENGINE_PROJECT, CreateMSBuildSettings("Publish")
```

Comment:
I cannot comment on the line below (lines 111-114). But shouldn't we add "net6.0" to building of agent project below? 

I.e. in
```
    foreach (var framework in new[] { "netcoreapp3.1", "net5.0" })
        MSBuild(AGENT_PROJECT, CreateMSBuildSettings("Publish")
            .WithProperty("TargetFramework", framework)
            .WithProperty("PublishDir", BIN_DIR + "agents/" + framework));
```

Reply:
I'll check this after I make the other changes. I tried to only publish where I determined experimentally that I needed to and it's possible that I didn't need to for net6.0.

Reply:
Well this is a mystery. I absolutely know we needed to do the Publish step at one time. Now I can comment out all that code and everything still works. It may be because I'm now using .NET  6.0 to host the build.

For now, I'll leave it as is and revisit in another issue.

Reply:
This reminds me about issue #1118. It seems to me that publishing in the same directory we use for compile output leads to confusion about where a particular file came from. What do you think?

Reply:
I think it is fine to separate them, as you suggest in #1118